clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -878,6 +879,7 @@ class SingleDeviceFunctionTracker {
         // Hence this test.
         // FIXME: We need to be more selective here, this can be hit by simply
         // having a kernel lambda with a lambda call inside of it.
+        Parent.SemaSYCLRef.addUserProvidedSYCLKernelFunction(CurrentDecl);
         KernelBody = CurrentDecl;
       }
```

Comment:
Suggest adding this once here common to both the 'if' and 'else' clauses.

Reply:
I moved it outside if else statement